src/routes/communityGroups/community_groups.contoller.js

```diff
@@ -0,0 +1,82 @@
+const fs = require("fs");
+const { promisify } = require("util");
+
+const { ERROR_CODES } = require("../../middleware/error-handler");
+
+const data = require("../../../db/communityGtoups.data.json");
+
+const writeFile = promisify(fs.writeFile);
+
+const dbPath = "db/communityGtoups.data.json";
+
+// return all community groups
+const listCommunityGroups = (req, res) => {
+  return res.status(200).json(data);
+};
+
+// get a community group by its id if it exists
+const getCommunityGroup = (req, res) => {
+  const record = data.filter(
+    record => record.id === parseInt(req.params.id)
+  )[0];
+
+  if (!record) {
+    throw new Error(
+      JSON.stringify({
+        status: ERROR_CODES.NOT_FOUND,
+        message: "Record not found"
+      })
+    );
+  }
+
+  return res.status(200).json(record);
+};
+
+// add a community group if the request has no errors
+const addCommunityGroup = async (req, res) => {
+  const id = data.length + 1;
+  const newCommunityGroupsData = [...data, { id, ...req.body }];
+  await writeFile(dbPath, JSON.stringify(newCommunityGroupsData));
+  return res.status(201).json({
```

Comment:
What is 201? Consider making a success status codes object like you did for the error codes

Reply:
Will do!